src/sage/quadratic_forms/bqf_class_group.py

```diff
@@ -330,6 +334,34 @@ def gens(self):
         """
         return [g.element() for g in self.abelian_group().gens()]
 
+    def _coerce_map_from_(self, other):
+        r"""
+        Return the natural projection map between two class groups
+        of binary quadratic forms when it is defined.
+
+        .. SEEALSO:: :class:`BQFClassGroupQuotientMorphism`
+
+        EXAMPLES::
+
+            sage: G = BQFClassGroup(-4*117117)
+            sage: H = BQFClassGroup(-4*77)
+            sage: proj = G.hom(H); proj  # implicit doctest
+            Coercion morphism:
+              From: Form Class Group of Discriminant -468468
+              To:   Form Class Group of Discriminant -308
+            sage: elt = G(BinaryQF(333, 306, 422)); elt
+            Class of 333*x^2 + 306*x*y + 422*y^2
+            sage: proj(elt)
+            Class of 9*x^2 + 4*x*y + 9*y^2
+        """
+        if not isinstance(other, BQFClassGroup):
+            return None
+        try:
+            proj = BQFClassGroupQuotientMorphism(other, self)
+        except (TypeError, ValueError):
+            return None
```

Comment:
```suggestion
            return super()._coerce_map_from_()
        try:
            proj = BQFClassGroupQuotientMorphism(other, self)
        except (TypeError, ValueError):
            return super()._coerce_map_from_()
```
This is better in case some super class decides to implement more general stuff. (Unlikely, but one never knows...)